navigation/navigation-runtime/src/webMain/kotlin/androidx/navigation/BrowserHistory.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.navigation
+
+import androidx.core.bundle.Bundle
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavController
+import androidx.navigation.NavGraph
+
+expect abstract class BrowserWindow
```

Comment:
It seems it's in both cases `org.w3c.dom.Window`, so we don't really need to introduce a new public type(alias)

Reply:
there is no common API between JS and Wasm targets, so we need it.

Reply:
See `compose/ui/ui/src/webCommonW3C`
@eymar @Schahen please confirm that we can share this code

Reply:
we cannot. it uses dirty hack with a reuse of the same source directory. it drives crazy the IDE

Reply:
I feel that having a new public/static `BrowserWindow` API is not really justified in this case. I'm even thinking that it might be better to just not have a common API between JS and WASM for this case

Reply:
you are right

Reply:
done